internal/marketplaces/subscriptions/service.go

```diff
@@ -36,51 +36,51 @@ import (
 
 // SubscriptionService defines operations on the subscriptions, as well as the
 // profiles and rules linked to subscriptions
+// It is assumed that all methods will be called in the context of a
+// transaction, and they take
 type SubscriptionService interface {
 	// Subscribe creates a subscription record for the specified project
 	// and bundle. It is a no-op if the project is already subscribed.
 	Subscribe(
 		ctx context.Context,
 		project types.ProjectContext,
 		bundle reader.BundleReader,
+		qtx db.ExtendQuerier,
 	) error
 	// CreateProfile creates the specified profile from the bundle in the project.
 	CreateProfile(
 		ctx context.Context,
 		project types.ProjectContext,
 		bundle reader.BundleReader,
 		profileName string,
-	) error
-	// CreateRuleTypes creates all rule types from the bundle in the project.
-	CreateRuleTypes(
-		ctx context.Context,
-		project types.ProjectContext,
-		bundle reader.BundleReader,
+		qtx db.ExtendQuerier,
 	) error
 }
 
 type subscriptionService struct {
 	profiles profsvc.ProfileService
 	rules    ruletypes.RuleTypeService
-	store    db.Store
 }
 
 // NewSubscriptionService creates an instance of the SubscriptionService interface
 func NewSubscriptionService(
 	profiles profsvc.ProfileService,
 	rules ruletypes.RuleTypeService,
-	store db.Store,
 ) SubscriptionService {
 	return &subscriptionService{
 		profiles: profiles,
 		rules:    rules,
-		store:    store,
 	}
 }
 
-func (s *subscriptionService) Subscribe(ctx context.Context, project types.ProjectContext, bundle reader.BundleReader) error {
+func (s *subscriptionService) Subscribe(
```

Comment:
Other change: I merged `Subscribe` and `CreateRuleTypes` together since they are always called together. Now that we have transactions, we can apply them atomically as well.